### PR TITLE
Use a single defaultWatcher per cluster

### PR DESCRIPTION
**Pull request by howardjohn**
Avoid multiple runs doing the same thing

**Please provide a description of this PR:**

### CODE REVIEW

File: pkg/revisions/default_watcher.go [hunk 1]
```diff
@@ -71,6 +72,12 @@ func (p *defaultWatcher) Run(stopCh <-chan struct{}) {
 		log.Errorf("failed to sync default watcher")
 		return
 	}
+	p.mu.Lock()
+	if p.started {
+		return
```
Comment (hzxuzhonghu) [inline pkg/revisions/default_watcher.go:77]: unlock

**Review by hzxuzhonghu — COMMENTED** [review]
I read the default watcher carefully, and find that it maybe better to just remove the queue from it since it is only reading a revision.  And keep instantiating it in each leader election call

File: pkg/revisions/default_watcher.go (hunk 1 above)
Comment (howardjohn) [reply pkg/revisions/default_watcher.go:88]: Thanks. refactored to use `defer`

**Comment by howardjohn** [comment]
> I read the default watcher carefully, and find that it maybe better to just remove the queue from it since it is only reading a revision. And keep instantiating it in each leader election call

The work it does is trivial, definitely. Do you know if k8s recommends/uses queues _always_, or only for non-trivial tasks?

I do still think its better to not instantiate multiple times per call though. That would cause multiple handlers on the informer still, which in theory could fall out of sync and cause weird behavior. It _should_ work fine but doesn't seem ideal.

**Comment by howardjohn** [comment]
I think we need to create it only if we are going to Run it. Creating the queue launches a goroutine and starts filling up the queue. if we never run it, the queue will grow unbounded

**Comment by howardjohn** [comment]
closing since its harder than expected

**Pull request closed without merging**